robottelo/orm.py

```diff
@@ -166,13 +169,33 @@ class attributes of ``Meta`` are metadata. Here is one way to instantiate
 
         User(name='Alice', supervisor=1, subordinate=[3, 4])
 
+    An entity object is useless if you are unable to use it to communicate with
+    a server. The solution is to provide a ``nailgun.config.ServerConfig`` when
+    instantiating a new entity. This configuration object is stored as an
+    instance variable named ``server_config_`` and used by methods such as
+    :meth:`robottelo.orm.Entity.path`.
+
+    1. If the ``server_config`` argument is specified, then that is used.
+    2. Otherwise, if :data:`robottelo.orm.DEFAULT_SERVER_CONFIG` is set, then
+       that is used.
+    3. Otherwise, call ``nailgun.config.ServerConfig.get()``.
+
     """
     # The id() builtin is still available within instance methods, class
     # methods, static methods, inner classes, and so on. However, id() is *not*
     # available at the current level of lexical scoping after this point.
     id = IntegerField()  # pylint:disable=C0103
 
-    def __init__(self, **kwargs):
+    def __init__(self, server_config=None, **kwargs):
+        # server_config > DEFAULT_SERVER_CONFIG > ServerConfig.get()
+        if server_config is not None:
+            self.server_config_ = server_config
```

Comment:
It is was I have imagined. In my personal opinion I would use `robottelo_server_config` instead of using a `_` in the end. Or maybe the python naming convention for private things like `_server_config`. 

I'm OK on merging this if you decide to continue with the adopted naming.